### ISSUE TITLE
Screen reader does not read selected list items on 'Market' dropdown control.

### ISSUE THREAD

**Issue by msftedad** [apache/superset#24856]
Screen reader does not read selected list items on 'Market' dropdown control.

#### How to reproduce the bug
1. Open URL: https://titanweb-testenv/superset/dashboard/268/ in Edge browser.
2. Navigate to the Market filter section appearing.
3. Turn on NVDA
4. Navigate and activate 'Market' dropdown control where list items appear.
5. Select list items.
6. Navigate to selected list items in dropdown control.
7. Observe that screen reader does not read selected list items on 'Market' dropdown control.

### Expected results
Content which is visually displayed must also be available to assistive technology users. If the content is dynamic (such as an error message), focus management and/or aria-live can expose it to a screen reader. If the content is static, be sure it is not hidden with CSS or using aria hidden.

### Actual results
Screen reader does not read selected list items on 'Market' dropdown control.

### Environment
OS: Windows_11 
Version: 22H2
OS Build: 22610.1
Browser: New edge (Version 105.0.1343.53)
URL:  https://titanweb-testenv/superset/dashboard/268/
Screen reader: NVDA (2021.3)

### Additional context
Screen reader does not provide complete information when focus lands on date control on Calander table where day -Monday, Tuesday etc. is not announced by screen reader.

**Attachments:**

https://github.com/apache/superset/assets/93735775/dcf5ac40-a694-4e0a-8348-f62987b4c861

**Comment by rusackas** [comment]
It's been a long while since this was opened. The issue's assignees might be able to get it onto an accessibility project board, but unless someone wants to contribute toward this soon, I think we'll likely close it since it's gone silent for so long, and it's more feature request than a bug (though we certainly understand how lack of accessibility features is an issue to many, and we want to see this area improve).

**Comment by rusackas** [comment]
Anyone object to closing this? It'd stay on the Accessibility project board in case anyone wants to volunteer to pick it up.
